concent_api/utils/tests/test_decorators.py

```diff
@@ -0,0 +1,222 @@
+        self.assertEqual(loaded_response.timestamp, self._parse_iso_date_to_timestamp("2017-12-31 00:00:00"))
+
+    def test_handle_errors_and_responses_should_return_http_response_if_it_has_been_passed_to_decorator(self):
+
+        request = self.request_factory.post("/dummy-url/", content_type = 'application/octet-stream')
+
+        @handle_errors_and_responses
+        def dummy_view_handle_http_response(_request, _message):
+            http_response = HttpResponse(status = 200)
+            return http_response
+
+        response = dummy_view_handle_http_response(request, self.client_auth)
+
+        self.assertIsInstance(response, HttpResponse)
+        self.assertEqual(response.status_code, 200)
+
+    def test_handle_errors_and_responses_should_return_http_response_if_view_passed_none(self):
+
+        request = self.request_factory.post("/dummy-url/", content_type = 'application/octet-stream')
+
+        @handle_errors_and_responses
+        def dummy_view_handle_none_response(_request, _message):
+            return None
+
+        response = dummy_view_handle_none_response(request, self.client_auth)  # pylint: disable=assignment-from-none
+
+        self.assertEqual(response.status_code, 204)
+        self.assertEqual(len(response.content), 0)
+
+    def test_handle_errors_and_responses_should_return_json_response_if_view_passed_dict(self):
+
+        request = self.request_factory.post("/dummy-url/", content_type = 'application/octet-stream')
+
+        @handle_errors_and_responses
+        def dummy_view_handle_dict(_request, _message):
+            return {}
+
+        response = dummy_view_handle_dict(request, self.client_auth)
+
+        self.assertEqual(response.status_code, 200)  # pylint: disable=no-member
+        self.assertIsInstance(response, JsonResponse)
+        self.assertEqual(json.loads(response.content), {})  # pylint: disable=no-member
+
+    def test_handle_errors_and_responses_should_return_http_response_if_view_passed_http_400_exception(self):
+
+        request = self.request_factory.post("/dummy-url/", content_type = 'application/octet-stream')
+
+        @handle_errors_and_responses
+        def dummy_view_handle_http_400_exception(_request, _message):
+            raise Http400('dummy')
+
+        response = dummy_view_handle_http_400_exception(request, self.client_auth)  # pylint: disable=assignment-from-no-return
+
+        self.assertEqual(response.status_code, 400)  # pylint: disable=no-member
+        self.assertIn('error', json.loads(response.content))  # pylint: disable=no-member
+
+    def test_handle_errors_and_responses_should_return_http_response_not_allowed_if_view_passed_it(self):
+
+        request = self.request_factory.post("/dummy-url/", content_type = 'application/octet-stream')
+
+        @handle_errors_and_responses
+        def dummy_view_handle_http_response_not_allowed(_request, _message):
+            return HttpResponseNotAllowed({}, status = 405)
```

Comment:
I do not mean to be picky, but we have agreed upon: `status=405` style formatting ;)